src/Core/OrganizationFeatures/AuthorizationHandlers/CollectionUserAuthorizationHandler.cs

```diff
@@ -0,0 +1,207 @@
+﻿using Bit.Core.Context;
+using Bit.Core.Entities;
+using Bit.Core.Repositories;
+using Microsoft.AspNetCore.Authorization;
+
+namespace Bit.Core.OrganizationFeatures.AuthorizationHandlers;
+
+public class CollectionUserAuthorizationHandler : BulkAuthorizationHandler<CollectionUserOperationRequirement, CollectionUser>
+{
+    private readonly ICurrentContext _currentContext;
+    private readonly ICollectionRepository _collectionRepository;
+    private readonly IOrganizationUserRepository _organizationUserRepository;
+
+    public CollectionUserAuthorizationHandler(ICurrentContext currentContext, ICollectionRepository collectionRepository, IOrganizationUserRepository organizationUserRepository)
+    {
+        _currentContext = currentContext;
+        _collectionRepository = collectionRepository;
+        _organizationUserRepository = organizationUserRepository;
+    }
+
+    protected override async Task HandleRequirementAsync(AuthorizationHandlerContext context,
+        CollectionUserOperationRequirement requirement,
+        ICollection<CollectionUser> resources)
+    {
+        switch (requirement)
+        {
+            case not null when requirement == CollectionUserOperation.Create:
+                await CanCreateAsync(context, requirement, resources);
+                break;
+            case not null when requirement == CollectionUserOperation.Delete:
+                await CanDeleteAsync(context, requirement, resources);
+                break;
+            case not null when requirement.Name == nameof(CollectionUserOperation.CreateForNewCollection):
+                await CanCreateForNewCollectionAsync(context, requirement, resources);
+                break;
+        }
+    }
+
+    /// <summary>
+    /// Ensure the acting user can create the requested <see cref="CollectionUser"/> resource(s).
+    /// </summary>
+    /// <remarks>
+    /// Checks that the following are all true:
+    /// - The target collection(s) exists 
+    /// - The acting user is an owner/admin AND the collection management setting is enabled
+    ///   OR
+    ///   The target collection(s) is manageable by the acting user. 
+    /// - The target user(s) exists and belongs to the same organization as the target collection
+    /// </remarks>
```

Comment:
I think this is too much detail for an xmldoc comment, and it's likely to become out of date. It's not really required given that the method itself is commented as well.

Reply:
Fair point about it becoming out of date; it started out as pseudo code for myself when writing the method and I figured it would also work as documentation/summary for the method so I moved it there. I agree that its a bit verbose, so I can remove it.

Reply:
Comments removed!